File: AndroidMVVMRxJavaRetrofit2/app/src/main/java/com/blackbirds/shakil/androidmvvmrxjavaretrofit2/Repositroies/MainRepo.java
package com.blackbirds.shakil.androidmvvmrxjavaretrofit2.Repositroies;

import androidx.lifecycle.MutableLiveData;

import com.blackbirds.shakil.androidmvvmrxjavaretrofit2.Common.Common;
import com.blackbirds.shakil.androidmvvmrxjavaretrofit2.Model.PostModel;
import com.blackbirds.shakil.androidmvvmrxjavaretrofit2.Network.APIService;

import java.util.List;

import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.disposables.CompositeDisposable;
import io.reactivex.functions.Consumer;
import io.reactivex.schedulers.Schedulers;

public class MainRepo {
    private static final String TAG = "MainRepo";

    private APIService apiService;
    private CompositeDisposable compositeDisposable = new CompositeDisposable();

    public MainRepo() {
        this.apiService = Common.getAPIService();
    }

    public MutableLiveData<List<PostModel>> getPostModelLiveData(){
        MutableLiveData<List<PostModel>> data = new MutableLiveData<>();

        compositeDisposable.add(apiService.getPosts()
        .subscribeOn(Schedulers.io())
        .observeOn(AndroidSchedulers.mainThread())
        .subscribe(postModels -> {
            if (postModels != null){
                data.setValue(postModels);
            }
        }));

        return data;
    }
}

File: KotlinMVVMRxJavaRetrofit2/app/src/main/java/com/blackbirds/shakil/kotlinmvvmrxjavaretrofit2/Repositories/MainRepo.kt
package com.blackbirds.shakil.kotlinmvvmrxjavaretrofit2.Repositories

import androidx.lifecycle.MutableLiveData
import com.blackbirds.shakil.kotlinmvvmrxjavaretrofit2.Common.Common
import com.blackbirds.shakil.kotlinmvvmrxjavaretrofit2.Model.PostModel
import com.blackbirds.shakil.kotlinmvvmrxjavaretrofit2.Network.APIService
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.disposables.CompositeDisposable
import io.reactivex.schedulers.Schedulers

class MainRepo {
    private val apiService: APIService
    private val compositeDisposable = CompositeDisposable()

    init {
        apiService = Common.getAPIService
    }

    val getPostModelLiveData: MutableLiveData<MutableList<PostModel>>
    get() {
        val data: MutableLiveData<MutableList<PostModel>> = MutableLiveData<MutableList<PostModel>>()
        compositeDisposable.add(apiService.getPosts()
            .subscribeOn(Schedulers.io())
            .observeOn(AndroidSchedulers.mainThread())
            .subscribe{postModels ->
                if (postModels != null){
                    data.value = postModels
                }
            })

        return data
    }
}